molgenis-data-import/src/main/java/org/molgenis/data/importer/ImportApi.java

```diff
@@ -0,0 +1,215 @@
+import java.net.URL;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+import static org.molgenis.data.importer.ImportApi.URI;
+
+@Controller
+@RequestMapping(URI)
+public class ImportApi
+{
+	private static final long serialVersionUID = 1L;
+	private static final Logger LOG = LoggerFactory.getLogger(ImportApi.class);
+
+	public static final String ID = "importer";
+	public static final String URI = "api/" + ID;
+
+	private ImportServiceFactory importServiceFactory;
+	private FileStore fileStore;
+	private FileRepositoryCollectionFactory fileRepositoryCollectionFactory;
+	private ImportRunService importRunService;
+	private ExecutorService asyncImportJobs;
+	private DataService dataService;
+
+	@Autowired
+	public ImportApi(ImportServiceFactory importServiceFactory, FileStore fileStore,
+			FileRepositoryCollectionFactory fileRepositoryCollectionFactory, ImportRunService importRunService,
+			DataService dataService)
+	{
+		this.importServiceFactory = importServiceFactory;
+		this.fileStore = fileStore;
+		this.fileRepositoryCollectionFactory = fileRepositoryCollectionFactory;
+		this.importRunService = importRunService;
+		this.dataService = dataService;
+		this.asyncImportJobs = Executors.newCachedThreadPool();
+	}
+
+	public ImportApi(ImportServiceFactory importServiceFactory, FileStore fileStore,
+			FileRepositoryCollectionFactory fileRepositoryCollectionFactory, ImportRunService importRunService,
+			DataService dataService, ExecutorService executorService)
+	{
+		this.importServiceFactory = importServiceFactory;
+		this.fileStore = fileStore;
+		this.fileRepositoryCollectionFactory = fileRepositoryCollectionFactory;
+		this.importRunService = importRunService;
+		this.dataService = dataService;
+		this.asyncImportJobs = executorService;
+	}
+
+	@RequestMapping(method = RequestMethod.POST, value = "/importByUrl")
+	@ResponseBody
+	public ResponseEntity<String> importFileByUrl(HttpServletRequest request,
+			@RequestParam("location") String fileLocation,
+			@RequestParam(value = "entityName", required = false) String entityName,
+			@RequestParam(value = "action", required = false) String action,
+			@RequestParam(value = "notify", required = false) Boolean notify) throws IOException
+	{
+		ImportRun importRun;
+		try
+		{
+			File tmpFile = fileLocationToStoredRenamedFile(fileLocation, entityName);
+			importRun = importFile(request, tmpFile, action, notify);
+		}
+		catch (Exception e)
+		{
+			LOG.error(e.getMessage());
+			return new ResponseEntity<>(e.getMessage(), HttpStatus.BAD_REQUEST);
+		}
+		return new ResponseEntity<>(
+				Href.concatEntityHref("/api/v2", importRun.getEntityMetaData().getName(), importRun.getIdValue()),
+				HttpStatus.CREATED);
+	}
+
+	@RequestMapping(method = RequestMethod.POST, value = "/importFile")
+	public ResponseEntity<String> importFile(HttpServletRequest request,
```

Comment:
I'm fine with statuscode + string as a minimalist way to get the result across, but for instance for the multiple entities upload we return a json object with more information so this is a bit of a diversification.